### PR TITLE
Renamed server.start to server.install

### LINKED ISSUE #11882: Rename server.start to server.install
ref: https://stackoverflow.com/questions/68969251/bokeh-server-start-not-showing-anything/68975581#68975581

One example of confusion linked above. The `Server.start` method merely installs Bokeh callbacks on an event loop, it does not start anything.  Propose to rename `start` to `install` to more accurately reflect what the method does, and add back a deprecated `start` method that delegates to `install`.

### PR DESCRIPTION
**Pull request by NaincyKumariKnoldus**
Hye! I have renamed server.start to server.install. Please review this!
fixes #11882

### CODE REVIEW

File: bokeh/_testing/plugins/file_server.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def where_is(self, path: str) -> str:
 @pytest.fixture(scope='session')
 def file_server(request: pytest.FixtureRequest) -> SimpleWebServer:
     server = SimpleWebServer()
-    server.start()
+    server.install()
```
Comment (bryevdv) [inline bokeh/_testing/plugins/file_server.py:121]: This is the `start` method of a Python built-in `SimpleWebServer`, not a Bokeh `Server` object (change should be reverted)

**Review by bryevdv — CHANGES_REQUESTED** [review]

**Comment by bryevdv** [comment]
Hi @NaincyKumariKnoldus Thanks for your PR. It looks like a blanket search-and-replace has swept up some unrelated places that should not be changed. I have left comments above. Additionally the actual primary task are:

* rename the existing `start` method on `BaseServer` to `install`
* add a new `start` method that calls `install` and also calls `deprecated` to issue a deprecation warning

and it does not appear either of those has been done. Let me know if there are further details I can clairfy

**Comment by NaincyKumariKnoldus** [comment]
Hey @bryevdv I have reverted the changes as you suggested. Review it now!

File: sphinx/docserver.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def serve_http():
     print(f"Visit http://localhost:{PORT}/en/latest/index.html to see plots\n")
 
     server = threading.Thread(target=serve_http)
-    server.start()
+    server.install()
```
Comment (bryevdv) [inline sphinx/docserver.py:94]: Here is another one that must be reverted. This is a Python `threading.Thread` not a Bokeh `Server`

File: bokeh/server/server.py [hunk 3]
```diff
@@ -442,7 +442,7 @@ def __init__(self, applications: Mapping[str, Application | ModifyDoc] | Applica
 
             http_server = HTTPServer(tornado_app, **http_server_kwargs)
 
-            http_server.start(opts.num_procs)
+            http_server.install(opts.num_procs)
```
Comment (bryevdv) [inline bokeh/server/server.py:445]: Another one. This is a tornado `HTTPServer` not a Bokeh `Server` or `BaseServer`

**Comment by bryevdv** [comment]
@NaincyKumariKnoldus there were some more instances that must be reverted. I'd like to point out that lots of tools and libraries have a "start" method. However, the only ones to be changes are the ones on Bokeh obejcts:

* `BaseServer`
* `Server`

So you need to do more than a blind search and replace. You need to actually examine all the places that a "start" is called, and verify that it is on a Bokeh class, and not somethings else.

Besides that, there is still the other wort that has not been done. It is described in the issue, and I have tried to describe a bit more detail above. To reiterate:

>  Additionally the actual primary task are:
> 
> * rename the existing start method on BaseServer to install
> * add a new start method that calls install and also issues a deprecation warning stating to use install instead

This PR will not be mergeable until that part is done.

**Comment by NaincyKumariKnoldus** [comment]
Hey @bryevdv Made the requested changes again. Review it!

**Comment by bryevdv** [comment]
Hi @NaincyKumariKnoldus You made the changes at the call sites, but you still have not made any changes to rename the function definition (which is why all the unit test jobs are failing). The PR will not be mergable until the function definition is rename, and tests are passing.  Please let me know if there is any question about this part of the work I can elaborate on.

**Comment by NaincyKumariKnoldus** [comment]
Hey @bryevdv again any changes is needed? what to change now

**Comment by bryevdv** [comment]
@NaincyKumariKnoldus I have already described the remaining work required in comments above, multiple times.

**Comment by bryevdv** [comment]
Closing due to lack of response and continued miscommunication.

**Pull request closed without merging**